### PR TITLE
Redesign Brave Wallet main layout introducing latest material 2 components

### PR DESCRIPTION
**Pull request by simoarpe**
## Resolves https://github.com/brave/brave-browser/issues/32197

https://github.com/brave/brave-core/assets/3308503/759ac54a-dfa6-4606-a188-b40fe4c8289d

* Remove logic for old Android versions below Nougat (< API level 24).
* Switch from old `ViewPager` to new and more performant `ViewPager2` class.
* Switch from deprecated `FragmentStatePagerAdapter` to new `FragmentStateAdapter`.

* Rename `Market` section to `Explore` section per Figma design.
* Implement `BottomNavigationViewIndicator` to display sections: Portfolio, NFTs, Activity, Accounts, Explore.
**⚠️ Note:** NFTs section will be removed from the bottom navigation view and it will be merged into the portfolio section in a follow up PR (Main umbrella Issue: https://github.com/brave/brave-browser/issues/32196)

Screenshot_2023-08-11_at_2_12_30_PM

### Bottom Navigation View

Remove tab layout in favor of [bottom navigation view](https://m2.material.io/components/bottom-navigation) with gradient indicator that animates when tapped.

https://github.com/brave/brave-core/assets/3308503/5751b96a-d5ee-42ee-a3a8-02a83d83639d

### App Bar Layout

Add [app bar layout](https://m2.material.io/components/app-bars-top) and rewrite layout to scroll correctly.

https://github.com/brave/brave-core/assets/3308503/e0bb2e6e-47ce-4836-b252-762a0241ca1e

### Tweak FAB behavior

FAB "follows" bottom navigation view

https://github.com/brave/brave-core/assets/3308503/407c5a50-b502-4c3e-9b73-91bd6f974bb1

### Banner layout

Improved layout with bottom margin and thin separator.

Screenshot_2023-08-11_at_2_27_07_PM

## Submitter Checklist:

- [x] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

* Open Brave Wallet.
* Observe the new bottom navigation view is displayed instead of the old tabs on top.
* Tap on its sections.
* Observe the pages switch accordingly.
* Navigate back to Portfolio section and scroll down
* Observe the bottom navigation view disappears.
* Observe the FAB button animation
* Observe the toolbar can disappear
* Scroll up
* Observe the bottom navigation view is shown again
* Observe the FAB "follows" the bottom navigation view
* Observe the toolbar reappears.

### CODE REVIEW

**Review by SergeyZhukovsky — APPROVED** [review]
++

File: android/java/org/chromium/chrome/browser/crypto_wallet/adapters/CryptoFragmentPageAdapter.java [hunk 1]
```diff
@@ -25,29 +24,36 @@
...
     private final List<String> mTitles;
 
     private PortfolioFragment mCurrentPortfolioFragment;
 
-    public CryptoFragmentPageAdapter(FragmentManager fm, Context context) {
-        super(fm, BEHAVIOR_RESUME_ONLY_CURRENT_FRAGMENT);
-        Resources resources = context.getResources();
+    public CryptoFragmentPageAdapter(FragmentActivity fm) {
+        super(fm);
+        Resources resources = fm.getResources();
```
Comment (Pavneet-Sing) [inline android/java/org/chromium/chrome/browser/crypto_wallet/adapters/CryptoFragmentPageAdapter.java:43]: nit: `fm` should be `activity`, no longer a `FragmentManager`

File: android/java/org/chromium/chrome/browser/crypto_wallet/fragments/PortfolioFragment.java [hunk 2]
```diff
@@ -447,9 +447,9 @@ private void updatePendingTxNotification() {
         Activity activity = getActivity();
         if (activity instanceof BraveWalletActivity)
             if (hasPendingTx())
-                ((BraveWalletActivity) activity).setPendingTxNotificationVisibility(View.VISIBLE);
+                ((BraveWalletActivity) activity).showPendingTxNotification(true);
             else
-                ((BraveWalletActivity) activity).setPendingTxNotificationVisibility(View.GONE);
+                ((BraveWalletActivity) activity).showPendingTxNotification(false);
```
Comment (Pavneet-Sing) [inline android/java/org/chromium/chrome/browser/crypto_wallet/fragments/PortfolioFragment.java:452]: Replace with ((BraveWalletActivity) activity).showPendingTxNotification(hasPendingTx());

**Review by Pavneet-Sing — APPROVED** [review]

File: android/java/org/chromium/chrome/browser/crypto_wallet/adapters/CryptoFragmentPageAdapter.java (hunk 1 above)
Comment (simoarpe) [reply android/java/org/chromium/chrome/browser/crypto_wallet/adapters/CryptoFragmentPageAdapter.java:43]: 👍

File: android/java/org/chromium/chrome/browser/crypto_wallet/fragments/PortfolioFragment.java (hunk 2 above)
Comment (simoarpe) [reply android/java/org/chromium/chrome/browser/crypto_wallet/fragments/PortfolioFragment.java:452]: 👍

File: android/java/res/layout/activity_brave_wallet.xml [hunk 3]
```diff
@@ -1,95 +1,143 @@
...
+                android:layout_width="56dp"
+                android:layout_height="56dp"
+                android:layout_marginEnd="16dp"
+                android:layout_marginBottom="16dp"
+                android:background="@drawable/ic_swap_bg"
+                android:contentDescription="@null"
+                android:elevation="10dp"
+                android:scaleType="center"
+                android:src="@drawable/ic_swap_icon" />
+        </LinearLayout>
```
[the remark this answers is not in the document]
Comment (simoarpe) [reply android/java/res/layout/activity_brave_wallet.xml:109]: Yes, I should try harder to make this a constraint layout. 
I'm going to tackle this task in the PR solving issue https://github.com/brave/brave-browser/issues/32256

[the remark this answers is not in the document]
Comment (Pavneet-Sing) [reply android/java/res/layout/activity_brave_wallet.xml:109]: Feel free to keep it as it is if it takes more of your time, just a nit.

**Review by mkarolin — APPROVED** [review]
`strings`++

**Pull request merged**